[9.99.x-prod][SRVLOGIC-228] Investigate why non-dev scenarios now requires JDBC configuration

wmedvede opened the pull request:
- kie-kogito-serverless-operator-379: Investigate why non-dev scenarios now requires JDBC configuration (#386)

    - Use "latest" image tag and namespaced images generation.

(cherry picked from commit df462a854fa1e879f72b90c02f2cae23df833008)

<!--

Welcome to the SonataFlow Operator! Before contributing, make sure to:

- Rebase your branch on the latest upstream main
- Link any relevant issues, PR's, or documentation
- Check that the commit message is concice and helpful:
    - When fixing an issue, add "Closes #<ISSUE_NUMBER>"
- Follow the below checklist 

-->

**Description of the change:**


**Motivation for the change:**


**Checklist**

- [X] Add or Modify a unit test for your change
- [X] Have you verified that tall the tests are passing?

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

rgdoliveira commented:
Just as a reminder, this change will need to be included in the `main` branch for kiegroup for future releases. I will keep that in mind.

wmedvede committed:
SRVLOGIC-SRVLOGIC-228: Investigate why non-dev scenarios now requires JDBC configuration

    - kie-kogito-serverless-operator-379: Investigate why non-dev scenarios now requires JDBC configuration (#386)

    - Use "latest" image tag and namespaced images generation.

(cherry picked from commit df462a854fa1e879f72b90c02f2cae23df833008)

ricardozanini commented:
@wmedvede can you run the `make generate-all` then `make vet fmt`? Or since go.work has been deleted those won't work anymore too?

wmedvede commented:
@ricardozanini @rgdoliveira 

**Quick question here**

When I try to execute this locally, I can see that for some e2e tests, that uses a sonataflow platform that should install an ephemeral DI and an ephemeral JS, the following images used for each service:

**DataIndex:**

the following image is picked:          
          image: >-
            **registry.redhat.io/openshift-serverless-1-tech-preview/logic-data-index-ephemeral-rhel8:1.32.0**

(this now fails, maybe the image is not published yet or my OpenShift local can't pick it, np)

**JobsService:**

the following image is picked:
          **image: 'quay.io/kiegroup/kogito-jobs-service-ephemeral-nightly:latest'**


And, in the operator code I can see that we have that different images are being picked up intentionally.

https://github.com/kiegroup/kogito-serverless-operator/blob/9.99.x-prod/controllers/platform/services/services.go#L105

https://github.com/kiegroup/kogito-serverless-operator/blob/9.99.x-prod/controllers/platform/services/services.go#L261

@ricardozanini Do we still want this?

ricardozanini commented:
@wmedvede we sure do since JS doesn't have an OSL image just yet.

wmedvede commented:
> @wmedvede can you run the `make generate-all` then `make vet fmt`? Or since go.work has been deleted those won't work anymore too?

They don't work anymore because of the missing go.work
